fuel-vm/src/interpreter/memory.rs

```diff
@@ -140,6 +140,13 @@ impl MemoryRange {
         Self(start..end)
     }
 
+    /// Splits range at given relative offset. Panics if offset > range length.
+    pub fn split_at_offset(self, at: usize) -> (Self, Self) {
```

Comment:
Maybe we should return an optional return type here instead of using a panicking assert?

Reply:
It seems you use this function only in one place, so maybe it is better to move this code there instead.

Reply:
The API here mirrors [`slice::split_at` from `core`](https://doc.rust-lang.org/std/primitive.slice.html#method.split_at), including the panic behavior. The code is written here instead of the call site since it's a useful abstraction, and IMO makes the code more readable.